### DISCUSSION TITLE
Support page inclusion using YAML frontmatter

### DISCUSSION THREAD

**Discussion by mschrader15** [zoni/obsidian-export#65] [Feature requests]
Hi @zoni ,

I would like to mark my notes for export inside of the YAML frontmatter instead of sticking them in a specific directory. Something like:

```
---
export: True
---
```
This would allow me to maintain the structure of my vault while also publishing only specific pages.

I would be happy to create a PR with this feature. Do you have any thoughts?

**Comment by zoni** [comment]
Hey @mschrader15, I've considered including such a feature in the past, so I would be happy to accept a PR implementing this functionality. A few quick notes:

1. Some thought will need to be given as to how users would opt-in to the feature. What CLI interface would you propose for this?
2. I wish to keep the core parsing logic as small as possible for maintainability. As such, this should be implemented entirely through [postprocessors](https://docs.rs/obsidian-export/latest/obsidian_export/type.Postprocessor.html) (hint: [PostprocessorResult::StopAndSkipNote](https://docs.rs/obsidian-export/latest/obsidian_export/enum.PostprocessorResult.html#variant.StopAndSkipNote) can be used here). #57 may serve as a good example.
3. I'd like to see tests included with an implementation. There should be some useful examples in https://github.com/zoni/obsidian-export/blob/main/tests/postprocessors_test.rs

**Reply by mschrader15** [reply]
This will be my first Rust PR so thanks for the notes/guidance!

Regarding the CLI interface, I was thinking something like `--front-matter-inclusion-key` where the description would be:

Only export files with the specified YAML key set to true. --front-matter-inclusion-key=export will only export notes with export: True in the YAML front-matter. All other notes will be ignored.

Let me know what you think!
